presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/UnaliasSymbolReferences.java

```diff
@@ -578,7 +578,7 @@ public PlanNode visitJoin(JoinNode node, RewriteContext<Void> context)
 
             if (node.getType().equals(INNER)) {
                 canonicalCriteria.stream()
-                        .filter(clause -> clause.getLeft().getType().equals(clause.getRight().getType()))
```

Comment:
Shouldn't we not skip canonicalization of the joinCriteria itself? [here](https://github.com/prestodb/presto/blob/9651f49cafa297f68460b04ad41b5d02695ac664/presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/UnaliasSymbolReferences.java#L813)

Reply:
This statement maps an equijoin condition from the build side to the probe side.  So if `a = b`, and `a` and `b` are both types where `equalValuesAreIdentical` is `true`, then there's no difference between them, and the optimizer can just say `b` is an alias for `a`.  Downstream, we will canonicalize all references to the output variables from this join.  So when we encounter downstream references to `b`, when we canonicalize it, we'll map it to `a`.  So in effect, `b` becomes `a` downstream.  This change prevents that from happening--`b` will still be `b` if its a tstz.

Reply:
Ahh i see i thought the canonicalization has already happened in [here](https://github.com/prestodb/presto/blob/9651f49cafa297f68460b04ad41b5d02695ac664/presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/UnaliasSymbolReferences.java#L813) whereas that was just replacement.